### ISSUE TITLE
Doesn't Create Database Entry

### ISSUE THREAD

**Issue by Felli** [kinderp/humhub-modules-services#2]
After trying this module I noticed that it doesn't create a database entry which then in turn breaks the site, is there a way of fixing this?

**Comment by kinderp** [comment]
Sorry, the module is not completed yet and i have installed it manually.
I'll update the readme to inform about that.
Can you give me the error that you have in your db? 
To solve the issue, you can try to create the tables that are used by the module.
You should set autoincrement for all id primary key.

CREATE TABLE `services_user_service` (
  `service_id` int(11) NOT NULL,
  `user_id` int(11) NOT NULL,
  `status` int(1) NOT NULL,
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

ALTER TABLE `services_user_service`
  ADD PRIMARY KEY (`service_id`,`user_id`);

---------------------------------------------------------------------------------------------

CREATE TABLE `services_user_message` (
  `message_id` int(11) NOT NULL,
  `user_id` int(11) NOT NULL,
  `is_originator` tinyint(4) DEFAULT NULL,
  `last_viewed` datetime DEFAULT NULL,
  `created_at` datetime DEFAULT NULL,
  `created_by` int(11) DEFAULT NULL,
  `updated_at` datetime DEFAULT NULL,
  `updated_by` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8;


ALTER TABLE `services_user_message`
  ADD PRIMARY KEY (`message_id`,`user_id`),
  ADD KEY `index_last_viewed` (`last_viewed`);

-------------------------------------------------------------------------------

CREATE TABLE `services_service` (
  `id` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

ALTER TABLE `services_service`
  ADD PRIMARY KEY (`id`);

------------------------------------------------------------------------------------

CREATE TABLE `services_message_entry` (
  `id` int(11) NOT NULL,
  `message_id` int(11) NOT NULL,
  `user_id` int(11) NOT NULL,
  `file_id` int(11) DEFAULT NULL,
  `content` text NOT NULL,
  `created_at` datetime DEFAULT NULL,
  `created_by` int(11) DEFAULT NULL,
  `updated_at` datetime DEFAULT NULL,
  `updated_by` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

ALTER TABLE `services_message_entry`
  ADD PRIMARY KEY (`id`),
  ADD KEY `index_user_id` (`user_id`),
  ADD KEY `index_message_id` (`message_id`);

-----------------------------------------------------------------------------------

CREATE TABLE `services_message` (
  `id` int(11) NOT NULL,
  `title` varchar(255) DEFAULT NULL,
  `created_at` datetime DEFAULT NULL,
  `created_by` int(11) DEFAULT NULL,
  `updated_at` datetime DEFAULT NULL,
  `updated_by` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

ALTER TABLE `services_message`
  ADD PRIMARY KEY (`id`),
  ADD KEY `index_updated` (`updated_at`),
  ADD KEY `index_updated_by` (`updated_by`);
------------------------------------------------------------------------------------------------------------------

**Comment by Felli** [comment]
Something to do with is what I'm getting.

```
SQL query:


CREATE TABLE services_user_service (
service_id int(11) NOT NULL,
user_id int(11) NOT NULL,
status int(1) NOT NULL,
created_at datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
updated_at datetime NOT NULL DEFAULT CURRENT_TIMESTAMP
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
MySQL said: Documentation

#1067 - Invalid default value for 'created_at' 
```

I don't know how much help this will be but that's it.

**Comment by kinderp** [comment]
What version of mysql have you?

SHOW VARIABLES LIKE "%version%";

Maybe the problem should be that the DEFAULT CURRENT_TIMESTAMP support for a DATETIME (datatype) was added in MySQL 5.6 and you have a minor version.

**Comment by Felli** [comment]
MySQL 5.5.38 is the version I'm using.

**Comment by kinderp** [comment]
You could try to use timestamp instead of datetime in the create table statement.

**Comment by Felli** [comment]
Well I think I've somewhat fixed some of the problem (As in for Humhub problems I've been having...) And I'm hoping that this also helps fix the problems I'm having with this module as well!

Edit:
The database entry creation issues still happens, but I'm guessing that's because this module is still under development.

**Comment by kinderp** [comment]
Yes is under development but, if you have time, could you paste here the error. I would realize where is the problem :)

**Comment by Felli** [comment]
It looks like an Outgoing Connection problem, I'm not sure exactly, just thinking this is causing the entries to not be created. I'll look more into this tho.
